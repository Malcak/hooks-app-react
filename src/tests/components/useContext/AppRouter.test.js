import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../../components/useContext/AppRouter';
import { UserContext } from '../../../components/useContext/UserContext';

describe('tests on the <AppRouter /> component', () => {
  const user = { id: '1234', name: 'Fernando', email: 'fernando@mail.com' };
  const wrapper = mount(
    <UserContext.Provider value={{ user: user }}>
      <AppRouter />
    </UserContext.Provider>
  );
  test('should display the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
