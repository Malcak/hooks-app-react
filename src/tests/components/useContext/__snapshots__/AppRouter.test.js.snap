// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on the <AppRouter /> component should display the component correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div>
    <h1>
      Home
    </h1>
    <hr />
    <pre>
      {
   "id": "1234",
   "name": "Fernando",
   "email": "fernando@mail.com"
}
    </pre>
  </div>
</div>
`;
